### DISCUSSION TITLE
An error occurred when lammps loaded the deepmd plugin.

### DISCUSSION THREAD

**Discussion by linkx-stu** [deepmodeling/deepmd-kit#2597] [Q&A]
We installed the software through this command：
          sh deepmd-kit-2.2.2-cuda11.6_gpu-Linux-x86_64.sh
    
    And we reinstall lammps:
          cmake -D PKG_PLUGIN=ON -D PKG_KSPACE=ON -D LAMMPS_INSTALL_RPATH=ON -D BUILD_SHARED_LIBS=yes -D CMAKE_INSTALL_PREFIX=${deepmd_root} -D CMAKE_INSTALL_LIBDIR=lib -D CMAKE_INSTALL_FULL_LIBDIR=${deepmd_root}/lib ../cmake
    
But running lammps gives an error：
 LAMMPS (23 Jun 2022 - Update 2)
  using 1 OpenMP thread(s) per MPI task
Open of file /public/home/tm_lkx/deepmd-kit/lib/deepmd_lmp/dpplugin.so failed: /public/home/tm_lkx/deepmd-kit/lib/deepmd_lmp/./././libtensorflow_framework.so.2: undefined symbol: _ZN6snappy20RawUncompressToIOVecEPKcmPK5iovecm
Loaded 0 plugins from /public/home/tm_lkx/deepmd-kit/lib/deepmd_lmp
Open of file /public/home/tm_lkx/deepmd-kit/lib/deepmd_lmp/libdeepmd_lmp.so failed: /public/home/tm_lkx/deepmd-kit/lib/deepmd_lmp/libdeepmd_lmp.so: undefined symbol: 
_ZN9LAMMPS_NS5Error4_allERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEiN3fmt6v8_lmp17basic_string_viewIcEENSA_17basic_format_argsINSA_20basic_format_contextINSA_8appenderEcEEEE

**Comment by njzjz** [comment]
The compiler or compile flags mismatch, which causes the symbols mismatch.

There are two options here:

(1) use the exactly same compiler and flags as the plugin;

(2) recompile the plugin against the C library.

**Reply by linkx-stu** [reply]
I installed lammps with intel compiler. What compiler is deepmd-kit compiled by default?

**Reply by njzjz** [reply]
Conda packages are all compiled by conda compilers.
https://anaconda.org/anaconda/gxx_linux-64
